packages/core/src/lib/pagination/position.mts

```diff
@@ -0,0 +1,45 @@
+type Change<T> = {
+  previous: T;
+  current: T;
+};
+
+type PageInfo = {
+  active: Change<number>;
+  total: number;
+  pageSize: number;
+};
+
+/**
+ * Given information about a page, decides the next position at which the active
+ * item should be rendered in the page.
+ */
+type PositionReducer = (info: PageInfo, pointer: number) => number;
+
+/**
+ * Creates the next position for the active item considering a finite list of
+ * items to be rendered on a page.
+ */
+export const finite: PositionReducer = ({ pageSize, total, active }) => {
+  const middle = Math.floor(pageSize / 2);
+  return total <= pageSize || active.current < middle
+    ? active.current
+    : active.current >= total - middle
+    ? active.current + pageSize - total
+    : middle;
```

Comment:
Let's rewrite that with `if`/`else` instead of nested ternary. (same comment for `infinite`)